app.py

```diff
@@ -53,11 +55,49 @@ def inject_configurations():
 # Routes
 @app.route('/')
 @login_required
-def index():  # put application's code here
+def index():
     current_year = get_current_year()
+    year = db.session.query(Year).filter_by(year=current_year).first()
     user = db.session.query(User).filter_by(email=session['email']).first()
-    courses_teacher = db.session.query(Course).filter_by(year=current_year).join(Teacher).filter(Teacher.user_id == user.id).all()
-    return render_template("home.html", user=user, courses=courses_teacher, evaluations=user.evaluations)
+    researcher = db.session.query(Researcher).filter_by(user_id=user.id).first()
+    teacher = db.session.query(Teacher).filter_by(user_id=user.id).first() if user.is_teacher else None
+
+    # Teacher data
+    courses_teacher = []
+    researcher_supervised = []
+
+    # Researcher data
+    researcher_courses = []
+    researcher_current_course = []
+    researcher_evaluations = []
+
+    if teacher:
```

Comment:
```suggestion
    if user.is_teacher :
       ...
    else : 
       researcher = db.session.query(Researcher).filter_by(user_id=user.id).first()
       ...
```

No need to go fetch the `teacher` in the database because you don't use it later. And you can check if the user is a researcher here in the condition as suggested in the comment above.

Reply:
I'm going to have to add an if condition in the else if the user is neither a teacher nor a researcher

Reply:
That is a possibility. You can use an if/elif/else checking on the session flags for example